source/templates/add_step_sidebar_step.slim

```diff
@@ -1,7 +1,7 @@
 .step
 	button.select.angular-animated[ng-click="selected()"] = svg("plus")
 	.step-content
-		.icon[ng-style="{'background-image': (step | normalizedStepIconURL | backgroundImagePropertyValue)}"]
+		.icon[ng-style="{'background-image': (addStepService.stepIconLoadedStatesByStepIds[step.id] ? (step | normalizedStepIconURL | backgroundImagePropertyValue) : null)}"]
```

Comment:
`addStepService.isIconLoaded(step.id) && (step | normalizedStepIconURL | backgroundImagePropertyValue)`

Reply:
and shouldn't we use a default image until the icon is loaded?

Reply:
I did that in the stylesheet instead: when there is no background image defined, the `:before` element below will be exposed, which contains a loading spinner.